.gitlab-ci.yml

```diff
@@ -25,18 +26,25 @@ default:
         CUSTOM_CI_BUILDS_DIR: "/usr/WS1/$$USER/gitlab-runner-builds-dir"
         # Note: the above will not work with /usr/workspace, you must specify WS1 or WS2
 
+## Job Specifications
 .test-core:
     extends: .lc-variables
     variables:
         PYTHON: "/usr/bin/python3"
         debug: t
         FLUX_TESTS_LOGFILE: t
     script:
+        - pwd
+        - echo $CI_DIRECTORY
+        - cd ${CORE_BUILD_DIR}
+        - pwd
         - lstopo --of xml >$(hostname).xml
         - export FLUX_HWLOC_XMLFILE=$(pwd)/$(hostname).xml
-        - !reference ['.build-core', 'script']
-        - cd ${CORE_BUILD_DIR}
         - make -j $(nproc) check
+        ## The following is teardown to make sure the xml file doesn't get reused
+        ## but maybe it's unnecessary. Subsequent tests will run under new shells...
+        - rm ${FLUX_HWLOC_XMLFILE}
+        - unset FLUX_HWLOC_XMLFILE
```

Comment:
@grondo any comments on this? I think the `unset` is probably necessary at the end of the script but removing the XML file might be useful cleanup.

Reply:
*unnecessary

Reply:
I don't have a strong opinion. Seems like keeping around the XML file might be useful in case something goes wrong. I'm not sure about the `unset` because I don't really understand how environment variables are inherited across gitlab pipelines.

Reply:
Ok, I'll just keep it as is for now, but I might play around with it more later.